src/components/Registration.jsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Card
+} from "@mui/material";
+import React, { useState, useEffect } from "react";
+import { StyledCardContent, StyledButton, StyledForm, StyledTitle, Heading, StyledTextField } from '../styles/styled';
+
+export const Registration = ({ onRegistration }) => {
+  const [user, setUser] = useState({
+    username: "",
+    email: "",
+    password: "",
+    passwordConfirm: "",
+  });
+
+  const [isValidForm, setIsValidForm] = useState(false);
+
+  useEffect(() => {
+    if (validateEmail(user.email) && user.password && onConfirmPassword()) {
+      setIsValidForm(true);
+    } else {
+      setIsValidForm(false);
+    }
+  }, [user]);
+
+  const validateEmail = (email) => {
+    return /^\S+@\S+\.\S+$/.test(email);
+  };
```

Comment:
drobnost - funkce je "pure" (https://en.wikipedia.org/wiki/Pure_function) (=nezávisí na ničem mimo svůj scope) a tak může být definovaná mimo komponentu. Takhle se vytvoří nová funkce každý render komponenty, kdežto by jí stačilo definovat jednou

Reply:
to samý pro "extractUsername "